rest-service/manager_rest/resource_manager.py

```diff
@@ -1178,15 +1176,11 @@ def _execute_system_workflow(self,
         if should_queue:
             self.sm.put(execution)
             self._workflow_queued(execution)
-            return execution
-
-        workflow_executor.execute_workflow(
-            execution,
-            bypass_maintenance=bypass_maintenance,
-            handler=send_handler,
-        )
-
-        return execution
+            return execution, []
+        message = execution.render_message(
+            bypass_maintenance=bypass_maintenance)
+        db.session.commit()
+        return execution, [message]
```

Comment:
`_execute_system_workflow` should be not a thing of its own, but it should just use `prepare_executions` as other executions do. RD-3162 is about that.